src/es-router.js

```diff
@@ -229,34 +261,61 @@ class EsRouter {
     }, []);
   }
 
-  //actual routing function
-  path(route, isQueryParam) {
+  /**
+   * Build a new url for path changes
+   * We assume the route has already been verified as a predefined route here (or not)
+   *
+   * @param  {string} newPath - input path, e.g. /account/login
+   * @return {string}         - output ready for history push, e.g. /base/account/login?foo=bar
+   */
+  buildNewUrl(newPath) {
+    const paramArray = this.createParamString(this.queryParams);
+    const paramArrayString = paramArray.length ? `?${paramArray.join('&')}` : '';
+    // make sure the new url starts with a slash
+    const newUrlBase = this.base.match(/^\//) ? this.base : `/${this.base}`;
+    const newUrl = (`${newUrlBase}${newPath}${paramArrayString}`)
+      .replace(/\/{2,}/g, '/'); // dedup consecutive slashes
+    return newUrl;
+  }
+
+  /**
+   * Actual routing function
+   *
+   * @param  {string}  route        - New path to naviate to. Absolute path, not including base
+   * @param  {Boolean} isQueryParam - ???
+   * @param  {Boolean} initialLoad  - if true, skips push state/hash change and just fires event
+   */
+  path(route, isQueryParam, initialLoad = false) {
     if (!route) {return;}
     let newPath = route;
-
     let newPathObject = this.getPreDefinedRoute(newPath);
-
-    //if path didn't match and is in strict mode, go home
-    if (!newPathObject && !this.notStrictRouting) {
+    let redirect = false;
+    //if path didn't match and is in strict mode, go home (redirect)
+    if (!newPathObject && this.strictRouting) {
       newPath = this.home.route;
       newPathObject = this.home;
+      redirect = true;
     }
     //run all pre-move functions
     if (!isQueryParam) {
       this.startRouteChange(this.currentPathObject, newPathObject);
     }
 
-    //build new url
-    const paramArray = this.createParamString(this.queryParams);
-    const paramArrayString = paramArray.length ? `?${paramArray.join('&')}` : '';
-    const newUrl = `${this.base || ''}${newPath}${paramArrayString}`;
-
-    //set new url
-    if (this.useHash) {
-      this.wasChangedByUser = true;
-      window.location.hash = newUrl;
-    } else {
-      window.history.pushState(null, null, newUrl);
+    const newUrl = this.buildNewUrl(newPath);
+
+    //push new state to the window, but only if this is not the initial load
+    //otherwise we end up with two copies of the initial state in browser history
+    //
+    //ignoring this rule for strict routing seems to be the only option right now
+    //Need to come up with a working solution for "redirects"
+    //Using hash or pushState for redirects breaks the browser's back button
+    if (initialLoad === false || redirect === true) {
+      if (this.useHash) {
+        this.wasChangedByUser = true;
+        window.location.hash = newUrl;
+      } else {
+        window.history.pushState(null, null, newUrl);
```

Comment:
per MDN second arg should be empty string and maybe first arg should be empty object